pkg/agent/controller/agent.go

```diff
@@ -228,94 +304,167 @@ func (a *Controller) serviceImportLister(transform func(si *mcsv1a1.ServiceImpor
+	exportSpec, err := mcs.NewExportSpec(svc, localServiceExport, a.clusterID)
+	if err != nil {
+		a.handleServiceExportTransformError(err, localServiceExport,
+			"ExportSpecCreationFailed", "Error creating service export spec")
 
 		return nil, false
 	}
 
-	serviceImport := a.newServiceImport(svcExport.Name, svcExport.Namespace)
+	err = exportSpec.MarshalObjectMeta(&brokerServiceExport.ObjectMeta)
+	if err != nil {
+		a.handleServiceExportTransformError(err, localServiceExport,
+			"ExportSpecMarshalFailed", "Error marshaling service export spec to ObjectMeta")
 
-	serviceImport.Spec = mcsv1a1.ServiceImportSpec{
-		Ports:                 []mcsv1a1.ServicePort{},
-		Type:                  svcType,
-		SessionAffinityConfig: new(corev1.SessionAffinityConfig),
+		return nil, false
 	}
 
-	serviceImport.Status = mcsv1a1.ServiceImportStatus{
-		Clusters: []mcsv1a1.ClusterStatus{
-			{
-				Cluster: a.clusterID,
-			},
-		},
+	//serviceImport := a.newServiceImport(localServiceExport.Name, localServiceExport.Namespace)
+
+	//serviceImport.Spec = mcsv1a1.ServiceImportSpec{
+	//	Ports:                 []mcsv1a1.ServicePort{},
+	//	Type:                  svcType,
+	//	SessionAffinityConfig: new(corev1.SessionAffinityConfig),
+	//}
+	//
+	//serviceImport.Status = mcsv1a1.ServiceImportStatus{
+	//	Clusters: []mcsv1a1.ClusterStatus{
+	//		{
+	//			Cluster: a.clusterID,
+	//		},
+	//	},
+	//}
+
+	//if svcType == mcsv1a1.ClusterSetIP {
+	//	if a.globalnetEnabled {
+	//		ip, reason, msg := a.getGlobalIP(svc)
+	//		if ip == "" {
+	//			klog.V(log.DEBUG).Infof("Service to be exported (%s/%s) doesn't have a global IP yet", localServiceExport.Namespace, localServiceExport.Name)
+	//			// Globalnet enabled but serviceObj doesn't have globalIp yet, Update the status and requeue
+	//			a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace, corev1.ConditionFalse, reason, msg)
+	//
+	//			return nil, true
+	//		}
+	//
+	//		serviceImport.Spec.IPs = []string{ip}
+	//	} else {
+	//		serviceImport.Spec.IPs = []string{svc.Spec.ClusterIP}
+	//	}
+	//
+	//	serviceImport.Spec.Ports = a.getPortsForService(svc)
+	//	/* We also store the clusterIP in an annotation as an optimization to recover it in case the IPs are
+	//	cleared out when here's no backing Endpoint pods.
+	//	*/
+	//	serviceImport.Annotations[clusterIP] = serviceImport.Spec.IPs[0]
+	//}
+
+	a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace,
+		mcsv1a1.ServiceExportValid, corev1.ConditionFalse, "AwaitingSync",
+		"Awaiting sync of the ServiceExport to the broker")
+
+	klog.V(log.DEBUG).Infof("Returning ServiceExport: %#v", brokerServiceExport)
+
+	return brokerServiceExport, false
+}
+
+// check whether a broker's service export originates from the local cluster
+func (a *Controller) isRemoteServiceExportOwned(se *mcsv1a1.ServiceExport) bool {
+	return se.GetLabels()[lhconstants.LighthouseLabelSourceCluster] == a.clusterID
```

Comment:
See previous note on using annotations or labels. Cluster is only available in Labels so this might be the better choice.
There are probably additional locations where this comment would apply, but DRY ;-)